salt/modules/virt.py

```diff
@@ -324,57 +324,52 @@ def _parse_qemu_img_info(info):
 
 def _get_uuid(dom):
     '''
-    Return a uuid from the named vm
-
-    CLI Example:
-
-    .. code-block:: bash
-
-        salt '*' virt.get_uuid <domain>
+    Get uuid from a libvirt domain object.
     '''
-    return ElementTree.fromstring(get_xml(dom)).find('uuid').text
+    uuid = ElementTree.fromstring(dom.XMLDesc(0)).find('uuid').text
```

Comment:
Why are all these having to be redefined?

What changed? This has worked in the past.  Do you have a reason to not use the get_xml() function that already exists, and looks like it is doing the same thing?

I want to make sure we are not breaking other peoples environments.

Reply:
Thanks @gtmanfred, I'm pretty sure this will not break other peoples environments, few reasons: 

- the previous version of the code used to reference `dom` as string a "name" while it should be an actual domain object.
- all the other _get functions are expecting a domain object not a string and many functions are raising exceptions because of treating an object as a string, then failing with `ERROR: The VM "<libvirt.virDomain object at 0x7f24e9e73198>" is not present`

take a look at this one here for example:
```
    def _info(dom):
        '''
        Compute the infos of a domain
        '''
        raw = dom.info()
        return {'cpu': raw[3],
                'cputime': int(raw[4]),
                'disks': _get_disks(dom),
                'graphics': _get_graphics(dom),
                'nics': _get_nics(dom),
                'uuid': _get_uuid(dom),
                'loader': _get_loader(dom),
                'on_crash': _get_on_crash(dom),
                'on_reboot': _get_on_reboot(dom),
                'on_poweroff': _get_on_poweroff(dom),
                'maxMem': int(raw[1]),
                'mem': int(raw[2]),
                'state': VIRT_STATE_NAME_MAP.get(raw[0], 'unknown')}
 ```
`dom` is a domain object and not a domain name string.